apps/explorer-e2e/src/integration/node-switcher.cy.js

```diff
@@ -0,0 +1,82 @@
+const nodeErrorType = 'node-error-type';
+const nodeErrorMsg = 'node-error-message';
+const nodeId = 'node-url-0';
+const customNodeBtn = 'custom-node';
+
+context('Node switcher', function () {
+  before('visit home page', function () {
+    cy.intercept('GET', 'https://static.vega.xyz/assets/capsule-network.json', {
+      hosts: ['http://localhost:3028/query'],
+    }).as('nodeData');
+    cy.visit('/');
+    cy.wait('@nodeData');
+  });
+
+  describe('form validations and responses', function () {
+    beforeEach('open node selector', function () {
+      cy.getByTestId('git-network-data').within(() => {
+        cy.getByTestId('link').click({ force: true });
+      });
+    });
+    it('node data is displayed', function () {
+      cy.getByTestId('node-row').should('have.length.at.least', 1);
+
+      cy.getByTestId('node-row')
+        .eq(0)
+        .within(() => {
+          cy.getByTestId(nodeId)
+            .should('exist')
+            .and('have.attr', 'aria-checked', 'true');
+          cy.get('label').should('have.text', Cypress.env('networkQueryUrl'));
+          cy.getByTestId('response-time-cell')
+            .find('.font-mono')
```

Comment:
suggestion (non blocking): Do you need the font-mono selector? Seems a bit brittle to rely on that being present. Same for the other assertions

Reply:
Agreed. I've changed the data-testid so I don't need to find the font-mono selector anymore